Room info update endpoint

jagerman opened the pull request:
Adds `PUT /room/{token}` endpoint to accept a new name, description, and default_{read,write,upload} settings.

Also fixes some incorrect returned values (including fields with `false` values instead of omitting them), and exposes the default_read/write/upload values to room info for moderators.

jagerman committed:
Add PUT endpoint to modify room details

jagerman committed:
Room info fixes

The actual data we were returning for rooms didn't quite match what is
in api.yaml, and was missing the room defaults info.

Changes:

- Don't include `'moderator': True` unless we are a moderator
- Don't include `'admin': True` unless we are an admin
- Don't include room description when it is null
- Add default_read/default_write/default_upload fields to room info if
  we have mod permissions so that a moderator client can see the current
  value.
- Fix tests for the above.

majestrate commented on sogs/routes/rooms.py:
```diff
@@ -58,6 +66,65 @@ def get_rooms():
+
+
+@rooms.put("/room/<Room:room>")
+@auth.user_required
```
feel like we could have a `@auth.require_admin` for this

jagerman replied:
~~I'm not sure how to make that work through a decorator, because it isn't just global admins but also room admins (which means we need to be able to call `room.check_admin(g.user)`, and so need the `room` argument).~~

jagerman replied:
Added.

jagerman committed:
Add @auth.admin_required and .mod_required decorators

These make sure there is a user and that the user has room-admin or
room-moderator permissions.

majestrate commented on sogs/routes/auth.py:
```diff
@@ -123,6 +123,41 @@ def required_user_wrapper(*args, **kwargs):
+    specified) permission."""
+    require_user()
+    if not (room.check_admin(g.user) if admin else room.check_moderator(g.user)):
+        abort_with_reason(http.FORBIDDEN, 'This endpoint requires moderator permissions')
```
error string should contain 'admin' not 'moderator' when it's looking for admin not moderator

jagerman replied:
Fixed.

majestrate reviewed (approved):
look fine
